controllers/secretsync_controller_test.go

```diff
@@ -110,8 +401,15 @@ func TestReconcile(t *testing.T) {
 			}
 
 			_, err := ssc.Reconcile(context.Background(), req)
-			if !errors.Is(err, test.err) {
-				t.Fatalf("unexpected error: %v", err)
+			if test.expectedError {
+				if err == nil {
+					t.Fatalf("expected error but got nil")
+				}
+				if err.Error() != test.expectedErrorString {
+					t.Fatalf("expected error %s but got %s", test.expectedErrorString, err.Error())
+				}
+			} else if err != nil {
+				t.Fatalf("did not expect an error but got: %v", err)
 			}
```

Comment:
```suggestion
			if len(test.expectedErrorString) > 0 {
				if err == nil || err.Error() != test.expectedErrorString {
					t.Fatalf("expected error %q, got %q", test.expectedErrorString, err)
				}
			} else if err != nil {
				t.Fatalf("unexpected error: %v", err)
			}
```

We don't need 2 variables (bool and string) to define the required error cases.